streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -494,7 +494,9 @@ public void process(final Record<Integer, Integer> record) {
 
         // e2e latency = 10
         task.addRecords(partition1, singletonList(getConsumerRecord(0, 0L)));
-        task.process(10L);
+        time = new MockTime(0L, 10L, 0L);
```

Comment:
We have already a global `MockTime` object in this test that we should reused instead of creating a new one. This way, we can avoid to add `setTime` and keep the `Time` member variable `final`. -- Or is there a reason why we cannot reuse the global `MockTime` object?

Also, I don't really see any call to the newly added methods in this test, so why do we change this test method at all?

Reply:
The reason I had to change StreamTaskTest is because of following issue:
StreamTask -> process(wallClockTime) method updates processor context's system time in this method: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java#L685
AbstractProcessorContect has another systemTime field: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractProcessorContext.java#L48 which is updated in streamTask's process method. I removed this field from AbstractProcessorContect as we want to fetch the time from StreamTask. But the latency is measured [here](https://github.com/apache/kafka/blob/b9dfc196aaee91f4fa23fe6563261c6f2d01ac33/streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java#L284) which is streamTask's time filed now which is not updated.
I couldn't find any other way to change time field in StreamTask unless I add setTime method in it. If i don't change time in StreamTask object, subsequent metric tests fail as stream-time is not updated. 
Do you have recommendation about how i can fix it without adding setTime method?

Reply:
In L159, we define a global mock time object:
```
private MockTime time = new MockTime();
```

that is passed into `StreamTask`. You can call `time.sleep(...)` to advance the mock time.